crypto/src/encryption.rs

```diff
@@ -0,0 +1,370 @@
+use crate::errors::CryptoError;
+use crate::types::{Ciphertext, PrivateKey};
+
+use num_bigint::BigUint;
+
+use tracing::{debug, error};
+
+/// Encrypt a string using ElGamal encryption.
+///
+/// The string is split into u32 chunks and each chunk is encrypted.
+///
+/// # Arguments
+///
+/// * `input` - The string to encrypt
+/// * `private_key` - The private key
+///
+/// # Returns
+///
+/// * `Vec<u8>` - The encrypted data
+pub fn encrypt(input: String, private_key: &PrivateKey) -> Vec<u8> {
```

Comment:
Ideally we want to give option to not only encrypt messages passed as `String`, I recommend to change function signature to something like:

```suggestion
pub fn encrypt<T: AsRef<[u8]>>(input: T, private_key: &PrivateKey) -> Vec<u8> {
```

But also don't forget to change the function body to use `.as_ref()`.

Reply:
Thanks! Resolved